salt/modules/zypper.py

```diff
@@ -649,10 +649,10 @@ def version_cmp(ver1, ver2, ignore_epoch=False):
     return __salt__['lowpkg.version_cmp'](ver1, ver2, ignore_epoch=ignore_epoch)
 
 
-def list_pkgs(versions_as_list=False, **kwargs):
+def list_pkgs(versions_as_list=False, attr=None, **kwargs):
```

Comment:
> attr=None should be added to kwargs, so that the list_pkgs is the same on all platforms

While it is true that many modules share the `(versions_as_list=False, **kwargs)` signature, that's definitely not universal:

```
salt$ ack -A3 "def list_pkgs"
[...]

salt/modules/yumpkg.py
585:def list_pkgs(versions_as_list=False, **kwargs):
[...]

salt/modules/dpkg.py
134:def list_pkgs(*packages):
[...]

salt/modules/nix.py
183:def list_pkgs(installed=True,
184-              attributes=True):
[...]

salt/modules/aptpkg.py
1318:def list_pkgs(versions_as_list=False,
1319-              removed=False,
1320-              purge_desired=False,
1321-              **kwargs):  # pylint: disable=W0613
[...]

salt/modules/rpm.py
129:def list_pkgs(*packages):
salt/modules/pkgng.py
[...]

salt/modules/pkgng.py
347:def list_pkgs(versions_as_list=False,
348-              jail=None,
349-              chroot=None,
350-              root=None,
351-              with_origin=False,
352-              **kwargs):
[...]

salt/modules/freebsdpkg.py
255:def list_pkgs(versions_as_list=False, with_origin=False, **kwargs):
[...]
```

That's even more evident when looking at other methods, like `install`:
```
salt$ ack -A15 "def install\("
[...]

salt/modules/npm.py
63:def install(pkg=None,
64-            pkgs=None,
65-            dir=None,
66-            runas=None,
67-            registry=None,
68-            env=None,
69-            dry_run=False,
70-            silent=True):
[...]

salt/modules/pacman.py
438:def install(name=None,
439-            refresh=False,
440-            sysupgrade=False,
441-            pkgs=None,
442-            sources=None,
443-            **kwargs):
[...]

salt/modules/pkgng.py
658:def install(name=None,
659-            fromrepo=None,
660-            pkgs=None,
661-            sources=None,
662-            jail=None,
663-            chroot=None,
664-            root=None,
665-            orphan=False,
666-            force=False,
667-            glob=False,
668-            local=False,
669-            dryrun=False,
670-            quiet=False,
671-            reinstall_requires=False,
672-            regex=False,
673-            pcre=False,
[...]

```

So I got the impression that uniform method signatures was not a design choice here - I personally think making parameters more visible in platform-specific modules has more value. I can of course change this if requested by the Salt team.